server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -24,41 +25,60 @@
  */
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
-    private ExtensionsManager.RequestType requestType;
+    private final ExtensionsManager.RequestType requestType;
+    private final String uniqueId;
 
     public ExtensionRequest(ExtensionsManager.RequestType requestType) {
+        this(requestType, null);
+    }
+
+    public ExtensionRequest(ExtensionsManager.RequestType requestType, String uniqueId) {
         this.requestType = requestType;
+        this.uniqueId = uniqueId;
     }
 
     public ExtensionRequest(StreamInput in) throws IOException {
         super(in);
         this.requestType = in.readEnum(ExtensionsManager.RequestType.class);
+        this.uniqueId = in.readOptionalString();
```

Comment:
Why uniqueId be an optionalString here? We do get uniqueId with ExtensionRequest

Reply:
because Extension will not always pass uniqueId to Opensearch, we should cover the condition of uniqueId is null. And `optionalString` method can handle this part.

Reply:
ExtensionRequest should have uniqueId always. This will help us to remove the uniqueId null check in this PR

Reply:
For all the other existing use cases of `ExtensionRequest` we are asking for information that does not depend on the calling extension.  Requiring us to include the `uniqueId` would add unnecessary bandwidth to the transport request.

This would require changing 3 other request/response use cases.  If you want to open a new issue to discuss that, I'm ok with further discussion, but I think in this case making it optional is fine.

I would be open to changing the return type from the `getUniqueId()` getter from a nullable `String` to an `Optional<String>` and handling the null check internally if you think that's a better way of handling it.